src/codegen/wasm.zig

```diff
@@ -140,3 +185,27 @@ fn genCall(buf: *ArrayList(u8), decl: *Decl, inst: *Inst.Call) !void {
         .decl = target,
     });
 }
+
+fn genStore(buf: *ArrayList(u8), decl: *Decl, inst: *Inst.BinOp) !void {
+    const idx = decl.fn_link.wasm.?.getLocalidx(inst.lhs).?;
+
+    const writer = buf.writer();
+
+    // generate codegen for rhs before we store it
+    try genInst(buf, decl, inst.rhs);
+
+    try writer.writeByte(0x21); // local.set
+    try leb.writeULEB128(writer, idx);
+}
+
+fn genLoad(buf: *ArrayList(u8), decl: *Decl, inst: *Inst.UnOp) !void {
+    const idx = decl.fn_link.wasm.?.getLocalidx(inst.operand).?;
+    const writer = buf.writer();
+
+    try writer.writeByte(0x20); // local.get
+    try leb.writeULEB128(writer, idx);
+}
+
+fn genAlloc(buf: *ArrayList(u8), decl: *Decl, inst: *Inst.NoOp) !void {
+    try decl.fn_link.wasm.?.locals.append(buf.allocator, &inst.base);
```

Comment:
I don't see any code that would guarantee that this next index has the type required for the given alloc instruction

Reply:
@ifreund Unfortunately the latest commit (6958a56) was the best I could come up with to make this work.